File: src/main/java/yaasl/server/demo/DatabaseConfig.java
package yaasl.server.demo;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Profile;
import yaasl.server.model.*;
import yaasl.server.persistence.*;

import javax.annotation.PostConstruct;
import java.util.Date;
import java.util.Optional;

import static org.apache.commons.lang3.time.DateUtils.addDays;
import static org.apache.commons.lang3.time.DateUtils.addMinutes;

@Configuration
@Profile("demo")
public class DatabaseConfig {

    @Autowired
    private LocationRepository locationRepository;

    @Autowired
    private PilotRepository pilotRepository;

    @Autowired
    private AircraftRepository aircraftRepository;

    @Autowired
    private FlightRepository flightRepository;

    @Autowired
    private UserRepository userRepository;

    @Autowired
    private AuthorityRepository authorityRepository;

    @Autowired
    private TemporaryAuthorityRepository temporaryAuthorityRepository;

    @PostConstruct
    public void initDatabase() {
        Date now = new Date();

        Flight flight = new Flight();
        flight.setStartLocation(locationRepository.findByIcao("LSZW"));
        flight.setLandingLocation(locationRepository.findByIcao("LSZW"));
        flight.setAircraft(getAircraft(1L));
        flight.setPilot1(getPilot(1L));
        flight.setPilot2(getPilot(2L));
        flight.setStartTime(addDays(now, -1));
        flight.setLandingTime(addMinutes(addDays(now, -1), 5));
        flight.setEditable(false);
        flight.setLocked(true);
        flightRepository.save(flight);

        flight = new Flight();
        flight.setStartLocation(locationRepository.findByIcao("LSZW"));
        flight.setLandingLocation(locationRepository.findByIcao("LSZW"));
        flight.setAircraft(getAircraft(2L));
        flight.setPilot1(getPilot(3L));
        flight.setPilot2(getPilot(4L));
        flight.setStartTime(now);
        flight.setLandingTime(addMinutes(now, 5));
        flight.setEditable(true);
        flight.setLocked(false);
        flightRepository.save(flight);

        flight = new Flight();
        flight.setStartLocation(locationRepository.findByIcao("LSZB"));
        flight.setLandingLocation(locationRepository.findByIcao("LSZB"));
        flight.setAircraft(getAircraft(3L));
        flight.setPilot1(getPilot(5L));
        flight.setPilot2(getPilot(6L));
        flight.setStartTime(addMinutes(addDays(now, -1), 5));
        flight.setLandingTime(addMinutes(addDays(now, -1), 25));
        flight.setEditable(false);
        flight.setLocked(true);
        flightRepository.save(flight);

        flight = new Flight();
        flight.setStartLocation(locationRepository.findByIcao("LSZB"));
        flight.setLandingLocation(locationRepository.findByIcao("LSZB"));
        flight.setAircraft(getAircraft(4L));
        flight.setPilot1(getPilot(7L));
        flight.setPilot2(getPilot(8L));
        flight.setStartTime(now);
        flight.setLandingTime(addMinutes(now, 5));
        flight.setEditable(true);
        flight.setLocked(false);
        flightRepository.save(flight);

        flight = new Flight();
        flight.setStartLocation(locationRepository.findByIcao("LSTB"));
        flight.setLandingLocation(locationRepository.findByIcao("LSTB"));
        flight.setAircraft(getAircraft(5L));
        flight.setPilot1(getPilot(7L));
        flight.setPilot2(getPilot(8L));
        flight.setStartTime(addMinutes(addDays(now, -1), 5));
        flight.setLandingTime(addMinutes(addDays(now, -1), 45));
        flight.setEditable(false);
        flight.setLocked(true);
        flightRepository.save(flight);

        flight = new Flight();
        flight.setStartLocation(locationRepository.findByIcao("LSTB"));
        flight.setLandingLocation(locationRepository.findByIcao("LSTB"));
        flight.setAircraft(getAircraft(6L));
        flight.setPilot1(getPilot(9L));
        flight.setStartTime(now);
        flight.setLandingTime(addMinutes(now, 5));
        flight.setEditable(true);
        flight.setLocked(false);
        flightRepository.save(flight);

        User fdl = userRepository.findByUsername("fdl");
        Authority authority = authorityRepository.findByName("fdl");
        TemporaryAuthority temporaryFdlAuthority = new TemporaryAuthority(now, fdl, authority);
        temporaryAuthorityRepository.save(temporaryFdlAuthority);
    }

    private Aircraft getAircraft(long id) {
        Optional<Aircraft> aircraft = aircraftRepository.findById(id);
        return aircraft.isPresent() ? aircraft.get() : null;
    }

    private Pilot getPilot(long id) {
        Optional<Pilot> pilot = pilotRepository.findById(id);
        return pilot.isPresent() ? pilot.get() : null;
    }

}

File: src/main/kotlin/yaasl/server/demo/DatabaseConfig.kt
package yaasl.server.demo

import org.apache.commons.lang3.time.DateUtils.addDays
import org.apache.commons.lang3.time.DateUtils.addMinutes
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Profile
import yaasl.server.model.Aircraft
import yaasl.server.model.Flight
import yaasl.server.model.Pilot
import yaasl.server.model.TemporaryAuthority
import yaasl.server.persistence.*
import java.util.*
import javax.annotation.PostConstruct

@Configuration
@Profile("demo")
class DatabaseConfig(
        private val locationRepository: LocationRepository,
        private val pilotRepository: PilotRepository,
        private val aircraftRepository: AircraftRepository,
        private val flightRepository: FlightRepository,
        private val userRepository: UserRepository,
        private val authorityRepository: AuthorityRepository,
        private val temporaryAuthorityRepository: TemporaryAuthorityRepository
        ) {

    @PostConstruct
    fun initDatabase() {
        val now = Date()

        var flight = Flight()
        flight.startLocation = locationRepository.findByIcao("LSZW")
        flight.landingLocation = locationRepository.findByIcao("LSZW")
        flight.aircraft = getAircraft(1L)
        flight.pilot1 = getPilot(1L)
        flight.pilot2 = getPilot(2L)
        flight.startTime = addDays(now, -1)
        flight.landingTime = addMinutes(addDays(now, -1), 5)
        flight.editable = false
        flight.locked = true
        flightRepository.save(flight)

        flight = Flight()
        flight.startLocation = locationRepository.findByIcao("LSZW")
        flight.landingLocation = locationRepository.findByIcao("LSZW")
        flight.aircraft = getAircraft(2L)
        flight.pilot1 = getPilot(3L)
        flight.pilot2 = getPilot(4L)
        flight.startTime = now
        flight.landingTime = addMinutes(now, 5)
        flight.editable = true
        flight.locked = false
        flightRepository.save(flight)

        flight = Flight()
        flight.startLocation = locationRepository.findByIcao("LSZB")
        flight.landingLocation = locationRepository.findByIcao("LSZB")
        flight.aircraft = getAircraft(3L)
        flight.pilot1 = getPilot(5L)
        flight.pilot2 = getPilot(6L)
        flight.startTime = addMinutes(addDays(now, -1), 5)
        flight.landingTime = addMinutes(addDays(now, -1), 25)
        flight.editable = false
        flight.locked = true
        flightRepository.save(flight)

        flight = Flight()
        flight.startLocation = locationRepository.findByIcao("LSZB")
        flight.landingLocation = locationRepository.findByIcao("LSZB")
        flight.aircraft = getAircraft(4L)
        flight.pilot1 = getPilot(7L)
        flight.pilot2 = getPilot(8L)
        flight.startTime = now
        flight.landingTime = addMinutes(now, 5)
        flight.editable = true
        flight.locked = false
        flightRepository.save(flight)

        flight = Flight()
        flight.startLocation = locationRepository.findByIcao("LSTB")
        flight.landingLocation = locationRepository.findByIcao("LSTB")
        flight.aircraft = getAircraft(5L)
        flight.pilot1 = getPilot(7L)
        flight.pilot2 = getPilot(8L)
        flight.startTime = addMinutes(addDays(now, -1), 5)
        flight.landingTime = addMinutes(addDays(now, -1), 45)
        flight.editable = false
        flight.locked = true
        flightRepository.save(flight)

        flight = Flight()
        flight.startLocation = locationRepository.findByIcao("LSTB")
        flight.landingLocation = locationRepository.findByIcao("LSTB")
        flight.aircraft = getAircraft(6L)
        flight.pilot1 = getPilot(9L)
        flight.startTime = now
        flight.landingTime = addMinutes(now, 5)
        flight.editable = true
        flight.locked = false
        flightRepository.save(flight)

        val fdl = userRepository.findByUsername("fdl")
        val authority = authorityRepository.findByName("fdl")
        val temporaryFdlAuthority = TemporaryAuthority().apply { date = now; user = fdl; this.authority = authority }
        temporaryAuthorityRepository.save(temporaryFdlAuthority)
    }

    private fun getAircraft(id: Long): Aircraft? {
        val aircraft = aircraftRepository.findById(id)
        return if (aircraft.isPresent) aircraft.get() else null
    }

    private fun getPilot(id: Long): Pilot? {
        val pilot = pilotRepository.findById(id)
        return if (pilot.isPresent) pilot.get() else null
    }

}
